.github/workflows/empty-string-warning.yml

```diff
@@ -0,0 +1,72 @@
+name: Warn for Empty Strings
+
+on:
+  pull_request:
+    paths:
+      - "**/*.ts"
+      - "**/*.tsx"
+
+permissions:
+  issues: write
+  pull-requests: write
+
+jobs:
+  check-empty-strings:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Find empty strings in TypeScript files
+        id: find_empty_strings
+        run: |
+          # Find empty strings and mark them explicitly
+          output=$(grep -Rn --include=\*.ts "''\|\"\"" --exclude-dir={node_modules,dist,out,tests} . || true)
```

Comment:
fucksake I wish I had seen this earlier but as I said before, me and bash not so great but I'm no quitter 😂

Took me way longer than I wanted it to, it's not 100% perfect neither but it's ~90-95% accurate.

- https://github.com/ubq-testing/bot-ai/pull/46

- seeing as it's a `commit comment` it's tied to each commit so if `""` is used in commit 1 and not resolved in commit 2 then commit 3 will retain the comment symbol but will not receive a comment as the change happened two commits ago, only once `""` are removed will the comment symbol disappear. (the 2nd last commit is proof of this, having an comment symbol but no actual comment)

- it's diffed from incoming commit > most recent, the reason why is because if `""` is used in commit 1 and unresolved in commit 2 then every single commit following will receive a comment (spamming if there are multiple) but it won't be tied to any file because the change is in commit 1 so it's just clutter and noise) 

- As for implementing an ignore switch, the wf runs on a push to PR so, it only checks for empty strings that are being pushed into the PR not ones that existed prior (unless edited). Disabling prior to running would be `.env` or `secret` based wouldn't it on a per repo basis? I can't think of how to disable it any other way of the top of my head